tools/generate-builders.ts

```diff
@@ -48,43 +48,78 @@ interface BuilderExtension {
 }
 /** Stores additional code that needs to be added to builders depending on their type */
 const buildersExtensions: { [key: string]: BuilderExtension } = {
+  Workflow: {
+    preValidate: `\n    data.expressionLang = data.expressionLang || 'jq';
+                        data.keepActive =  data.keepActive || false;`,
+  },
+  Exectimeout: {
+    preValidate: `\n    data.interrupt = data.interrupt || false;`,
+  },
+  Transition: {
+    preValidate: `\n    if (typeof data !== typeof '') (data as any).compensate = (data as any).compensate || false;`,
+  },
+  Onevents: {
+    preValidate: `\n    data.actionMode = data.actionMode || 'sequential';`,
+  },
   Callbackstate: {
-    preValidate: `\r\n    data.type = 'callback';`,
+    preValidate: `\n    data.type = 'callback';
+                        data.usedForCompensation = data.usedForCompensation || false;`,
   },
   Databasedswitch: {
-    preValidate: `\r\n    data.type = 'switch';`,
+    preValidate: `\n    data.type = 'switch';
+                        data.usedForCompensation = data.usedForCompensation || false;`,
   },
   Delaystate: {
-    preValidate: `\r\n    data.type = 'delay';`,
+    preValidate: `\n    data.type = 'delay';
+                        data.usedForCompensation = data.usedForCompensation || false;`,
   },
   Eventbasedswitch: {
-    preValidate: `\r\n    data.type = 'switch';`,
+    preValidate: `\n    data.type = 'switch';
+                        data.usedForCompensation = data.usedForCompensation || false;`,
   },
   Eventstate: {
-    preValidate: `\r\n    data.type = 'event';`,
+    preValidate: `\n    data.type = 'event';
+                        if (data.exclusive == null) data.exclusive = true;`,
   },
   Foreachstate: {
-    preValidate: `\r\n    data.type = 'foreach';
-                  \r\n    //FIXME https://github.com/serverlessworkflow/sdk-typescript/issues/95
-                  \r\n    data.usedForCompensation = data.usedForCompensation || false;`,
+    preValidate: `\n    data.type = 'foreach';
+                        //FIXME https://github.com/serverlessworkflow/sdk-typescript/issues/95
```

Comment:
If you pass by here please remove this fixme, otherwise I will do no problem

Reply:
done